.circleci/config.yml

```diff
@@ -0,0 +1,64 @@
+version: 2.1
+
+jobs:
+  build:
+    docker:
+      # see https://hub.docker.com/r/codeaches/openjdk
+      - image: codeaches/openjdk:12-jdk
+
+    working_directory: ~/repo
+
+    environment:
+      JVM_OPTS: -Xmx3200m
+      TERM: dumb
+
+    steps:
+      - checkout
+
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "build.gradle" }}
+            - v1-dependencies-
+
+      - run: gradle dependencies
+
+      - save_cache:
+          paths:
+            - ~/.gradle
+          key: v1-dependencies-{{ checksum "build.gradle" }}
+
+      - run: gradle build
```

Comment:
Just to clarify: by running `gradle build` these build steps will deploy the jar generated from this job and not the jar that's currently being committed to the repo, yeah?

Reply:
Yep, that's correct! Would it be better to deploy the jar it checks out from github? I wanted to protect against someone (me) forgetting to rebuild the jar before merging into master, but I guess this means the jar on master will always be different to that on production (even though they'll function in the same way), which seems wrong.

Reply:
I just read your comment on my next PR: https://github.com/lottetreg/httpserver/pull/3#issuecomment-512036474. Looks like I shouldn't be committing the jar file at all, so that answers my questions! :)

Follow up question: does it make sense to deploy the jar file to the server as opposed to deploying everything and having the server build the jar itself?

Reply:
The only pattern I've used (personally and in client work) is building the jar file in the CI pipeline and then deploying that to the server. 
Couple things to consider/think about: 
- What would happen if there's an error building the jar file (for whatever reason) if it's being built on the server? Would that leave production in a bad state?
If its being generated in the CI pipeline and it fails then presumably nothing would be deployed and you'd be able to remedy the problem and try run job/build steps again. 

How is it working today?

Reply:
Ok great! Yeah, right now I'm just deploying the jar file to the server from the CI pipeline, so the server doesn't have any building responsibilities. That's a very good point about risking the state of production if the server failed to build the jar.

Reply:
(It's totally OK to build and ship a jar on your CI server, just make sure it fails safely!)